libs/utils/energy.py

```diff
@@ -509,4 +511,30 @@ def report(self, out_dir, out_energy='energy.json'):
 
         return EnergyReport(channels_nrg, nrg_file, None)
 
+class Gem5EnergyMeter(_DevlibContinuousEnergyMeter):
+    def __init__(self, target, conf, res_dir):
+        super(Gem5EnergyMeter, self).__init__(target, res_dir)
+
+        power_sites = conf['channel_map'].values()
+        self._instrument = devlib.Gem5PowerInstrument(self._target, power_sites)
+
+    def reset(self):
+        self._instrument.reset()
+        self._instrument.start()
+
+    def _build_timeline(self, df):
+        # Power measurements on gem5 are performed not only periodically but also
+        # spuriously on OPP changes. Let's use the time channel provided by the 
+        # gem5 power instrument to build the timeline accordingly. 
+        for site, measure in df:
+            if measure == 'time':
+                meas_dur = df[site]['time']
+                break
+        timeline = np.zeros(len(meas_dur))
+        # The time channel gives the elapsed time since previous measurement
+        for i in range(1, len(meas_dur)):
+            timeline[i] = meas_dur[i] + timeline[i-1]
```

Comment:
Tiny nitpick here, dunno if it's a proper Python thing, but I find it better when there are whitespaces around arithmetic operands, i.e. `timeline[i - 1]`